code/modules/holomap/holomap.dm

```diff
@@ -0,0 +1,500 @@
+		Since maps aren't that (they are 255x255) the compromise is defaulting to render
+		starting from [9,9] up to [248,248] which is the center portion of the map with
+		a border of 8 tiles that get culled. This can be changed with `region_selection`;
+		Any region larger than 240 will have its edges culled to fit, regardless of the
+		region size it will be centered in the `bigassicon`. Happy Viewing.
+	*/
+	var/region_width = (region_selection[3] - region_selection[1]) + 1
+	var/region_height = (region_selection[4] - region_selection[2]) + 1
+
+	var/region_x_start = (region_width > 240) ? 9 : region_selection[1]
+	var/region_y_start = (region_height > 240) ? 9 : region_selection[2]
+
+	var/region_x_finish = (region_width > 240) ? region_x_start + 239 : region_selection[3]
+	var/region_y_finish = (region_width > 240) ? region_y_start + 239 : region_selection[4]
+
+	var/render_x_start = (region_width > 240) ? -8 : 128 - (region_width / 2)
+	var/render_y_start = (region_width > 240) ? -8 : 128 - (region_height / 2)
+
+	var/icon_to_use = 'icons/effects/holomap_parts.dmi'
+
+	for(var/x_index in region_x_start to region_x_finish)
+		for(var/y_index in region_y_start to region_y_finish)
+			var/turf/T = locate(x_index, y_index, zlevel_rendered)
+			var/area/A = get_area(T)
+			if(A.type != /area/space && A.type != /area/space/nearstation)
+				if(location_areas[A])
+					location_areas[A][2] += T
+				else
+					location_areas[A] = list(A, list(T))
+			var/W = FALSE
+			for(var/obj/structure/window/full/WW in T.contents)
+				W = TRUE
+				break
+			for(var/obj/effect/spawner/window/WW in T.contents)
+				W = TRUE
+				break
+
+			/*
+				This works by using 2x2 sprites from 'icons/effects/holomap_parts.dmi' to construct the holomap;
+				We use different colors that we will swap later on for floors and walls (BLACK-->floor;WHITE-->wall)
+				The walls can either be a full 2x2 pixel box, 1x2 pixel wall, or a 3/4 pixel corner depending on their surroundings...
+				All the following logic is to determine what icon to use.
+			*/
+			//this is what all the logic is for, setting this variable
+			var/icon_state_to_use
+			//as well as this since it modifies the wall sprite
+			var/dir_to_use = 2
+
+			//the following variables are set conditionally on the state of the turf
+			//so we can render the right sprite for this turf
+			var/neighbor_space = FALSE
+			var/list/neighbors_alien = list()
+			var/list/neighbors_solid = list()
+
+	//--------------{{{ DISCOVERY PHASE }}}------------------------------
+
+			for(var/i in GLOB.cardinal)
+				var/turf/found_turf = get_step(T, i)
+				var/area/found_area = get_area(found_turf)
+				var/found_window = FALSE
+
+				for(var/obj/structure/window/full/WW in found_turf.contents)
+					found_window = TRUE
+					break
+				for(var/obj/effect/spawner/window/WW in found_turf.contents)
+					found_window = TRUE
+					break
+
+				for(var/j in GLOB.diagonals)
+					var/turf/found_s_turf = get_step(T, j)
```

Comment:
why is this for loop inside the GLOB.cardinal loop? It's not effected by cardinal stuff from what it looks like, you're potentially running this for loop 3 times extra PER turf (of which there at 65 thousand) which results in 16 iterations per turf.

Reply:
some walls that are diagonal to space don't get rendered correctly as full squares and look bad as a result, this check is to pick up on this case.
![StrongDMM_CWPrh4FTg4](https://github.com/ParadiseSS13/Paradise/assets/16618648/2a15b3cc-66c1-4365-9b66-9124fddc56c4)

Reply:
yes, but why does this need to be inside the current for loop?

Reply:
TRUEE, I guess I just misplaced. will fix